src/app/file-manager/progress-tab/progress-tab.component.ts

```diff
@@ -0,0 +1,172 @@
+
+import { DownloadItem } from '@app/file-manager/models/download-item';
+import { DownloadsReducer, DownloadState, downloadInitialState } from '@app/file-manager/reducers/downloads.reducer';
+
+import { UploadItem } from '@app/file-manager/models/upload-item';
+import { UploadsReducer, UploadState, uploadInitialState } from '@app/file-manager/reducers/uploads.reducer';
+
+import { ExportToGCSItem } from '@app/file-manager/models/export-to-gcs-item';
+import { ExportToGCSReducer, ExportToGCSState, exportToGCSInitialState } from '@app/file-manager/reducers/export-to-gcs.reducer';
+
+import { ExportToS3Item } from '@app/file-manager/models/export-to-s3-item';
+import { ExportToS3Reducer, ExportToS3State, exportToS3InitialState } from '@app/file-manager/reducers/export-to-s3.reducer';
+
+import { Observable } from 'rxjs/Observable';
+import { BehaviorSubject } from 'rxjs/BehaviorSubject';
+
+import { AppState } from '@app/file-manager/reducers';
+import { EntityStatus } from '@app/file-manager/models/entity-status';
+
+@Component({
+  selector: 'app-progress-tab',
+  templateUrl: './progress-tab.component.html',
+  styleUrls: ['./progress-tab.component.scss']
+})
+export class ProgressTabComponent implements OnInit, AfterViewInit, OnDestroy {
+
+  @Input('filter') set setFilter(filter) {
+    this.applyFilter(filter);
+  }
+
+  @Input('source') source;
+
+  @ViewChild(MatPaginator) paginator: MatPaginator;
+  @ViewChild(MatSort) sort: MatSort;
+  dataSource = new MatTableDataSource([]);
+  displayedColumns = ['name', 'size', 'status', 'progress', 'actions'];
+
+  downloads: Observable<DownloadState>;
+  uploads: Observable<UploadState>;
+  exportsToGCS: Observable<ExportToGCSState>;
+  exportsToS3: Observable<ExportToS3State>;
+
+  private subscription: ISubscription;
+
+  constructor(private store: Store<AppState>, private spinner: NgxSpinnerService, private zone: NgZone, ) {
+
+    this.downloads = store.select('downloads');
+    this.uploads = store.select('uploads');
+    this.exportsToGCS = store.select('exportToGCS');
+    this.exportsToS3 = store.select('exportToS3');
+
+  }
+
+  ngOnInit() {
+
+    switch (this.source) {
+
+      case 'Down':
+        this.subscription = this.downloads.subscribe(
+          state => {
+            this.processChanges(state);
+          }
+        );
+        break;
+
+      case 'Up':
+        this.subscription = this.uploads.subscribe(
+          state => {
+            this.processChanges(state);
+          }
+        );
+        break;
+
+      case 'GCS':
+        this.subscription = this.exportsToGCS.subscribe(
+          state => {
+            this.processChanges(state);
+          }
+        );
+        break;
+
+      case 'S3':
+        this.subscription = this.exportsToS3.subscribe(
+          state => {
+            this.processChanges(state);
+          }
+        );
+        break;
+
+    }
+
+  }
+
+  private processChanges(state) {
+    const pending = Object.values(state.pending.items);
```

Comment:
Only `inProgress` and `failed` are used below, can the unused ones be removed?

Reply:
Not for now. We want to keep this code until we get feedback from actual users and confirm, or not, the approach implemented.